src/lib_ccx/hardsubx.c

```diff
@@ -43,7 +43,7 @@ int hardsubx_process_data(struct lib_hardsubx_ctx *ctx, struct lib_ccx_ctx *ctx_
 		fatal(EXIT_READ_ERROR, "Video Stream not found!\n");
 	}
 
-	ctx->codec_ctx = ctx->format_ctx->streams[ctx->video_stream_id]->codec;
+	ctx->codec_ctx = ctx->format_ctx->streams[ctx->video_stream_id]->codecpar;
```

Comment:
Hi @prateekmedia,

Thanks for looking into this but I am afraid that this doesn't work on the Debian side. I cherry-picked this patch on top of the 0.94 release and it still failed to build. Here are the logs:

https://paste.ubuntu.com/p/vwZNJGWpy6/

Let me know if I'm missing something? Thanks in advance. :)

Reply:
What commands are you using to build this?

Reply:
On top of the regular stuff, just using `--enable-ocr --enable-hardsubx --with-rust=no`

Reply:
So if you follow the above logs, you can see that translates to:
`cd linux && ./configure --build=x86_64-linux-gnu --prefix=/usr --includedir=\${prefix}/include --mandir=\${prefix}/share/man --infodir=\${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --disable-option-checking --disable-silent-rules --libdir=\${prefix}/lib/x86_64-linux-gnu --runstatedir=/run --disable-maintainer-mode --disable-dependency-tracking --enable-ocr --enable-hardsubx --with-rust=no`

and then `make -j8`

Reply:
I believe this issue is not related to ffmpeg Compatibility, this pr only fixes ffmpeg 5.0 with respect to the related issue hence this can be a new issue. I will try to look this by setting up a ubuntu/Debian machine and compiling this with said arguments.

Reply:
@utkarsh2102 can you check now?

Reply:
@prateekmedia: nice, I shall do so and let you know! :D

Reply:
Fixed it!

Reply:
@utkarsh2102 I have pushed a commit with the same fix you proposed, can you check now. I don't know why these errors not coming in my system I am using latest Ubuntu 22.04 for compiling

Reply:
> @utkarsh2102 I have pushed a commit with the same fix you proposed, can you check now.

@prateekmedia: lovely, with that patch above, I can finally get the build working!

> I don't know why these errors not coming in my system I am using latest Ubuntu 22.04 for compiling

Aha! That's why. Ubuntu 22.04 (Jammy) doesn't have FFmpeg 5.x. 22.04 has 4.4.2. If you upgrade to the devel release, you'll be able to reproduce those issues. Or perhaps use Debian unstable.

Reply:
@prateekmedia remember FFmpeg 5 (not 4) is what we care about :-)
We don't update FFmpeg often, so when we do it's to latest.

Reply:
got it, will use devel version. If this is building then marking this as resolved.